distribution/errors.go

```diff
@@ -78,7 +78,7 @@ func TranslatePullError(err error, ref reference.Named) error {
 		switch v.Code {
 		case errcode.ErrorCodeDenied:
 			// ErrorCodeDenied is used when access to the repository was denied
-			newErr = errors.Errorf("repository %s not found: does not exist or no pull access", reference.FamiliarName(ref))
+			newErr = errors.Errorf("pull access denied for %s, repository does not exist or may require 'docker login'", reference.FamiliarName(ref))
```

Comment:
thanks - I looked through `image_routes.go` on the engine side, and `image_create.go` and `image_pull.go` on the cli side, as well as some general grepping & digging, and didn't discover anything that was matching on the existing error message.